papers/michael_lange/michael_lange.rst

```diff
@@ -641,11 +641,39 @@ flops" are performed.>*
 Integration with YASK
 ~~~~~~~~~~~~~~~~~~~~~
 
-*<YASK, and why it is so great.>* **[CITE]**
+As already explained, Devito is based upon actual compiler technology, and its
+backend presents a highly modular structure, with each transformation pass
+taking as input an AST and returning a new, different AST. One of the reasons
+behind this software engineering strategy, which is clearly more challenging than a
+template-based solution, is to ease the integration of external tools. One such
+tool is the YASK stencil optimizer **[CITE]**. We are currently integrating
+YASK within the DLE; YASK will replace some (but not all) of the existing DLE
+passes.
+
+The DLE passes are organized in a hierarchy of classes. Each class represents a
+specific code transformation pipeline; each stage of the pipeline manipulates
+ASTs. Integrating YASK becomes then a conceptually simple task, which boils
+down to three actions: (i) adding a new transformation pipeline to the DLE;
+(ii) adding a new array type, to ease storage layout transformations and data
+views (YASK employs a data layout different than the conventional row-major
+format); (iii) creating the proper Python bindings in YASK so that Devito can
+drive the code generation process. At the moment of writing, some progress has
+already been made: 1) Devito ASTs can now automatically be translated into YASK
+ASTs through an extremely simple tree visitor; 2) a Devito-generated acoustic
```

Comment:
I agree that the detailed description of progress might be a bit too much here, and I suspect that we will need to trim some of this stuff due to space restrictions at a later stage anyway. However, in the meantime, this is a good place to start from, so merging with intent to fix later.